### ISSUE TITLE
Move a task to another project

### ISSUE THREAD

**Issue by georgeweb-git** [Dolibarr/dolibarr#23230]
### Feature Request

Move a task to another project

### Use case

In the lifetime of projects it often happens, that a reorganization is done, and it becomes necessary to reorganize the projects as well. Especially, some open tasks have to be moved from one project to another.

### Suggested implementation

On the modify task, there can be added the list of projects that are assigned to the contacts, in order to choose which project you want to move the task.

![move_task_on_dolibarr](https://user-images.githubusercontent.com/25401194/207733486-aebc480e-a72b-495f-9ef8-324645e0a1e0.JPG)

### Suggested steps
No response

**Comment by milenmk** [comment]
I have a PR to add "Clone" and "Delete" as mass actions to project's tasks, but not sure if it will be included in v17 OR v17.01: [https://github.com/Dolibarr/dolibarr/pull/23143](https://github.com/Dolibarr/dolibarr/pull/23143)

However, you can use the changes to implement them into your installation.

**Comment by georgeweb-git** [comment]
Thanks @milenmk 

There’s no sign of this feature( [changelog](https://github.com/Dolibarr/dolibarr/blob/develop/ChangeLog)) in version 17

**Comment by milenmk** [comment]
It is not added yet. As I wrote, I am not sure if this will be added to v17 or v17.0.1. However, you can look at the PR/commit and change your files to have this feature. Changes will work for v16 also.

**Comment by isamuse2** [comment]
Such a feature allowing to move a task would indeed be very helpful, in addition to the possibility of cloning of deleting a tasK.

**Comment by edupulpillo** [comment]
I also reported this request in the issue  [#19177](https://github.com/Dolibarr/dolibarr/issues/19177) with the title:
Is not possible to select a different parent project in a task to allow to move tasks between different projects

**Comment by edupulpillo** [comment]
is the feature already added in 17.0.4 ?

**Comment by milenmk** [comment]
> is the feature already added in 17.0.4 ?

No. But I've looked at the code and I can see that it is present in the latest v18 release.

**Comment by edupulpillo** [comment]
I do not find the option in 18.0.4 GUI.

**Comment by Francewhoa** [comment]
For those wondering, according to those two change log for versions 18 and 19, this move task feature is not yet added:

* https://github.com/Dolibarr/dolibarr/blob/18.0/ChangeLog

* https://github.com/Dolibarr/dolibarr/blob/19.0/ChangeLog

If it was added to v18 or 19, does anyone have the change log line number or the commit number?

**Comment by Francewhoa** [comment]
I suggest considering this same workflow as Redmine below. It is very easy and quick for end-users to move a task to another project or sub-project.

## Two Steps

1. Click on this **Edit** link

   ![Step-1---Move-ticket---Screenshot-799---V2](https://github.com/Dolibarr/dolibarr/assets/119942/58221774-1191-4f8c-86cf-03446c56ab42)


2. Then, using this **Project** field, select any destination project or sub-project to your liking. Only the those you presently have access to are listed. Next, click on this **Submit** button. Done. Enjoy :)
   ![Step-2---Move-ticket---Screenshot-800---v2](https://github.com/Dolibarr/dolibarr/assets/119942/19f69ee6-ce4f-4292-ac74-1dc4f5e32014)




## Demo

Try a demo at https://alphanodes.com/redmine-demo 

After log-in above, this is an example of a task/issue available at the time of this comment at https://redmine-demo.alphanodes.net/issues/176



## Contributing

Any volunteer for a patch? Both me and the Ubertus.Org team would be happy, as volunteer, to contribute testing and documentation, if needed.



## Cloning

About cloning a task, I suggest creating a new Github ticket for this. Because both moving or cloning would be useful.

**Comment by edupulpillo** [comment]
ANy plan of the version of Dolibarr to incorporate this feature?

**Comment by JonBendtsen** [comment]
Okay, so for those who want to have this function, you can sort of and sort of not move it to another project - you can CLONE a task.

# Task tab under a project
<img width="745" height="904" alt="Image" src="https://github.com/user-attachments/assets/2515d4df-001c-4c53-97ea-c8b701a1c953" />

## select a task
<img width="654" height="247" alt="Image" src="https://github.com/user-attachments/assets/88bda5ab-3b4b-4915-86c2-c9516cfcd28b" />

## select clone and confirm 
<img width="478" height="243" alt="Image" src="https://github.com/user-attachments/assets/09ca5d9b-f51d-42fa-b5ee-ebd0e658f4e9" />

## In the dropdown of projects, select the target project and click yes
<img width="478" height="238" alt="Image" src="https://github.com/user-attachments/assets/57323099-5ba5-4d57-817a-acb71c890254" />

# The task is then cloned to the new project. Not moved, Cloned!
<img width="519" height="978" alt="Image" src="https://github.com/user-attachments/assets/ba986089-2c70-4e6f-b14e-73bb3d2ae3da" />

# Notice my task numbers
- TK2511-0005 - first created on the test sub project
- TK2511-0006 - clone of TK2511-0005 to jons_test_project
- TK2511-0007 - clone of TK2511-0006 - back to the test sub project

**Comment by Francewhoa** [comment]
Thanks for this clone @JonBendtsen 💚

I like the clone instead of move. One benefit of clone is that it reduces the risk of broken links. I mean, for example, if someone goes to the original Task URL, this link will still work. End-users can decide how they want to redirect people to the new cloned Task. For example, but not limited to, closing the original Task, then add a comment with a link to the new cloned Task. For example, "This Task has moved. Find new Task at `<URL>`".

**Comment by edupulpillo** [comment]
Is this feature already commited in Dolibarr ?

**Comment by JonBendtsen** [comment]
> Is this feature already commited in Dolibarr ?

I think the clone functionality is already there, and I think it is better than move

**Comment by edupulpillo** [comment]
Yes, and the clone functionality is already available in 22.0.4, that is good, so thank you for the contribution.
Maybe a good option to add an additional  function to copy/move the attached files to one task to another task?  Then after it if is not necessary to maintain the initial task, just is needed to delete it in the initial project.
Also maybe need to take in consideration the asociated objects to the original task, and check if is possible to link it also to the new cloned task.

**Comment by JonBendtsen** [comment]
> Yes, and the clone functionality is already available in 22.0.4, that is good, so thank you for the contribution. Maybe a good option to add an additional function to copy move the attached files to one task to another task? Then after it if is not necessary to maintain the initial task, just is needed to delete it in the initial project.

that is a really good idea, so good I think it should have it's own issue. My recommendation is that you create such an issue
